lib/srv/desktop/rdp/rdpclient/src/rdpdr/scard.rs

```diff
@@ -49,6 +48,21 @@ pub struct ScardBackend {
     pin: String,
 }
 
+impl std::fmt::Debug for ScardBackend {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ScardBackend")
+            .field("client_handle", &self.client_handle)
+            .field("contexts", &self.contexts)
+            .field("uuid", &self.uuid)
+            .field("cert_der", &util::vec_u8_debug(&self.cert_der))
+            // Important we don't leak key_der to the logs.
+            .field("key_der", &util::vec_u8_debug(&self.key_der))
+            // Out of an abundance of caution, don't leak the PIN to the logs.
+            .field("pin", &util::str_debug(&self.pin))
```

Comment:
I think we want to have access to pin in logs, it's sometimes the only way to log in and leaking it should not cause any security problems